cmds/config

```diff
@@ -1,37 +1,70 @@
 #!/bin/bash
 
+config_generate_dir() {
+    local sstate_dir=${1}
+
+    mkdir -p "${TOP}/${BUILD_DIR}/${CONF_DIR}"
+
+    pushd "${TOP}/${BUILD_DIR}" >/dev/null
+
+    if [ -n "${sstate_dir}" ]; then
+        if [ ! -e ../${sstate_dir} ]; then
+            mkdir ../${sstate_dir}
+        fi
+
+        ln -s ${sstate_dir} sstate-cache
+    else
+        if [ ! -e ../sstate-cache ]; then
+            mkdir ../sstate-cache
+        fi
+
+        ln -s ../sstate-cache
+    fi
+
+    for d in layers downloads ccache mirror ${CERTS_DIR}; do
+        if [ ! -e ../${d} ]; then
+            mkdir ../${d}
+        fi
+
+        ln -s ../${d}
+    done
+
+    popd >/dev/null
+}
+
 config_generate_auto() {
     local branch=${1}
-    local sstate_dir=${2}
 
-    if [  -e "${CONF_DIR}/auto.conf" ]; then
+    if [  -e "${TOP}/${BUILD_DIR}/${CONF_DIR}/auto.conf" ]; then
```

Comment:
Hmm, do we want to regenerate the config if someone re-runs config with an existing BUILD_ID? One approach is to cowardly ask if the really want to regenerate the config instead of blindly stopping on an existing build, just in case someone isn't paying attention (like that would never happen ^_^ ).

Reply:
:D, agreed, asking the question is the safer.